server/ServerFSM.cpp

```diff
@@ -2167,62 +2164,53 @@ sc::result WaitingForMPGameJoiners::react(const JoinGame& msg) {
+        while (player_name.compare(0, ai_prefix.size(), ai_prefix) == 0)
+            player_name.erase(0, ai_prefix.size());
+        if(player_name.empty())
+            player_name = "_";
 
-            std::string new_player_name = player_name;
+        std::string new_player_name = player_name;
 
-            bool collision = true;
-            std::size_t t = 1;
-            while (collision &&
-                   t <= m_lobby_data->m_players.size() + server.Networking().GetCookiesSize() + 1)
-            {
-                collision = false;
-                roles.Clear();
-                if (!server.IsAvailableName(new_player_name) || server.IsAuthRequiredOrFillRoles(new_player_name, roles)) {
-                    collision = true;
-                } else {
-                    for (std::pair<int, PlayerSetupData>& plr : m_lobby_data->m_players) {
-                        if (plr.second.m_empire_name == new_player_name) {
-                            collision = true;
-                            break;
-                        }
+        bool collision = true;
+        std::size_t t = 1;
+        while (collision &&
+               t <= m_lobby_data->m_players.size() + server.Networking().GetCookiesSize() + 1)
+        {
+            collision = false;
+            roles.Clear();
+            if (!server.IsAvailableName(new_player_name) || server.IsAuthRequiredOrFillRoles(new_player_name, roles)) {
+                collision = true;
+            } else {
+                for (std::pair<int, PlayerSetupData>& plr : m_lobby_data->m_players) {
+                    if (plr.second.m_empire_name == new_player_name) {
+                        collision = true;
+                        break;
                     }
                 }
-
-                if (collision)
-                    new_player_name = player_name + std::to_string(++t); // start alternative names from 2
             }
 
-            if (collision) {
-                player_connection->SendMessage(ErrorMessage(str(FlexibleFormat(UserString("ERROR_PLAYER_NAME_ALREADY_USED")) % original_player_name),
-                                                            true));
-                server.Networking().Disconnect(player_connection);
-                return discard_event();
-            }
+            if (collision)
+                new_player_name = player_name + std::to_string(++t); // start alternative names from 2
+        }
 
-            if (!player_connection->HasAuthRole(Networking::ROLE_CLIENT_TYPE_PLAYER)) {
-                player_connection->SendMessage(ErrorMessage(UserStringNop("ERROR_CLIENT_TYPE_NOT_ALLOWED"), true));
-                server.Networking().Disconnect(player_connection);
-                return discard_event();
-            }
+        if (collision) {
+            player_connection->SendMessage(ErrorMessage(str(FlexibleFormat(UserString("ERROR_PLAYER_NAME_ALREADY_USED")) % original_player_name),
```

Comment:
This isn't great as the error message is being formatted server-side, so a player with a different language set won't see the message in their chosen language...

Reply:
This collision check code is the same as `MPLobby::JoinGame`.
To implement client-side formatting there should be something like mentioned here: https://github.com/freeorion/freeorion/pull/2393#discussion_r261364566

Reply:
I generally assume the such strings should / would be sent as a stringtable key or enum and the string formatted client-side, but not using the `[[KEY]]` type markup.

Reply:
What if send array of string instead of one string in the ERROR_MSG message?